### PR TITLE
WIP curvefs/metaserver: now we support rocksdb storage

### PR DESCRIPTION
**Pull request by Wine93**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curvefs/src/metaserver/dentry_storage.h [hunk 1]
```diff
@@ -60,60 +62,50 @@ class DentryStorage {
...
+    MetaStatusCode Clear();
 
-    MetaStatusCode List(const Dentry& dentry,
-                        std::vector<Dentry>* dentrys,
-                        uint32_t limit) override;
+ private:
+    std::string DnetryKey(const Dentry& dentry, bool ignoreTxId = false);
 
-    MetaStatusCode HandleTx(TX_OP_TYPE type, const Dentry& dentry) override;
+    std::string DnetryValue(const Dentry& dentry);
```
Comment (Wine93) [inline curvefs/src/metaserver/dentry_storage.h:87]: typo

File: curvefs/src/metaserver/metaserver.cpp [hunk 2]
```diff
@@ -126,6 +127,7 @@ void Metaserver::InitPartitionOption(std::shared_ptr<S3ClientAdaptor> s3Adaptor,
 }
 
 void Metaserver::Init() {
+    ::curvefs::metaserver::storage::Init();
```
Comment (Wine93) [inline curvefs/src/metaserver/metaserver.cpp:130]: rm

Comment (Wine93) [reply curvefs/src/metaserver/metaserver.cpp:130]: 调用 `InitStorage`

File: curvefs/src/metaserver/metastore.cpp [hunk 3]
```diff
@@ -261,8 +283,8 @@ MetaStatusCode MetaStoreImpl::CreatePartition(
         return status;
     }
 
-    partitionMap_.emplace(partition.partitionid(),
-                          std::make_shared<Partition>(partition));
+    auto partition = std::make_shared<Partition>(partition, kvStorage_)
```
Comment (Wine93) [inline curvefs/src/metaserver/metastore.cpp:286]: 这里重复了

File: curvefs/src/metaserver/metastore.cpp [hunk 4]
```diff
@@ -147,52 +148,62 @@ bool MetaStoreImpl::Load(const std::string& pathname) {
...
+        container->emplace(std::to_string(partitionId), value);
     }
 
-    auto iterator = std::make_shared<PartitionIteratorType>(
-        ENTRY_TYPE::PARTITION, 0, container);
-    return iterator;
+    auto partitionId = partition->GetPartitionId();
+    auto iterator = std::make_shared<ContainerType>(container);
+    return std::make_shared<IteratorWrapper>(
+        ENTRY_TYPE::PARTITION, partitionId, iterator);
```
Comment (Wine93) [inline curvefs/src/metaserver/metastore.cpp:166]: partitionId 这里是 0

File: curvefs/src/metaserver/storage/storage_fstream.cpp [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+        return false;
+    }
+
+    auto defer = absl::MakeCleanup([&dumpfile]() { dumpfile.Close(); });
+
+    auto iter = dumpfile.Load();
+    for (iter->SeekToFirst(); iter->Valid(); iter->Next()) {
+        auto ikey = iter->Key();
+        auto value = iter->Value();
+        auto ukey = iter->Key();
```
Comment (Wine93) [inline curvefs/src/metaserver/storage/storage_fstream.cpp:151]: 这里应该是 UseKey()

**Comment by Wine93** [comment]
recheck

**Comment by Wine93** [comment]
recheck

**Comment by YunhuiChen** [comment]
recheck

**Comment by Wine93** [comment]
recheck

**Pull request closed without merging**